subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+		"catalog-sync-token",
+		"connect-inject-token",
+		"mesh-gateway-token",
+	}
+	policies, _, err := consul.ACL().PolicyList(nil)
+	require.NoError(err)
+	for _, expected := range expectedPolicies {
+		found := false
+		for _, p := range policies {
+			if p.Name == expected {
+				found = true
+
+				// We assert that the policy doesn't have any namespace config
+				// in it because later that's what we're using to test that it
+				// got updated.
+				policy, _, err := consul.ACL().PolicyRead(p.ID, nil)
+				require.NoError(err)
+				require.NotContains(policy.Rules, "namespace")
+				break
+			}
+		}
+		require.True(found, "Did not find "+expected)
+	}
+
+	// Re-run the command with namespace flags. The policies should be updated.
+	cmd = Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	responseCode = cmd.Run(secondRunArgs)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	// Check that the policies have all been updated.
+	policies, _, err = consul.ACL().PolicyList(nil)
+	require.NoError(err)
+	for _, expected := range expectedPolicies {
+		found := false
+		for _, p := range policies {
+			if p.Name == expected {
+				found = true
+				policy, _, err := consul.ACL().PolicyRead(p.ID, nil)
+				require.NoError(err)
+				if expected == "connect-inject-token" {
+					// The connect inject token doesn't have namespace config,
+					// but does change to operator:write from an empty string.
+					require.Contains(policy.Rules, "operator = \"write\"")
+				} else {
+					require.Contains(policy.Rules, "namespace")
+				}
+				break
+			}
+		}
+		require.True(found, "Did not find "+expected)
+	}
+}
+
+// Test the auth method and acl binding rule created when namespaces are enabled
+// and there's a single consul destination namespace.
+func TestRun_ConnectInject_SingleDestinationNamespace(t *testing.T) {
+	t.Parallel()
+	k8s, testAgent := completeEnterpriseSetup(t, resourcePrefix)
+	defer testAgent.Stop()
+	setupK8sServiceAccount(t, k8s)
+	require := require.New(t)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	args := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-create-inject-auth-method",
+		"-enable-namespaces",
+		"-consul-inject-destination-namespace=dest",
+	}
+
+	responseCode := cmd.Run(args)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	bootToken := getBootToken(t, k8s, resourcePrefix)
+	consul, err := api.NewClient(&api.Config{
+		Address: testAgent.HTTPAddr,
+		Token:   bootToken,
+	})
+	require.NoError(err)
+
+	// Check the ACL auth method is created in the expected namespace.
+	authMethodName := "release-name-consul-k8s-auth-method"
+	methods, _, err := consul.ACL().AuthMethodList(&api.QueryOptions{
+		Namespace: "dest",
+	})
+	require.NoError(err)
+	found := false
+	for _, m := range methods {
```

Comment:
Alternatively, there should only be one auth method in the `dest` namespace, so you could check that the length of the list is 1 and that the single item is the correct auth method.

Reply:
Switched to
```go
			// Check the ACL auth method is created in the expected namespace.
			authMethodName := releaseName + "-consul-k8s-auth-method"
			namespaceQuery := &api.QueryOptions{
				Namespace: consulDestNamespace,
			}
			actMethod, _, err := consul.ACL().AuthMethodRead(authMethodName, namespaceQuery)
			require.NotNil(actMethod)
			require.Equal("kubernetes", actMethod.Type)
			require.Equal("Kubernetes AuthMethod", actMethod.Description)
			require.NotContains(actMethod.Config, "MapNamespaces")
			require.NotContains(actMethod.Config, "ConsulNamespacePrefix")

			// Check the binding rule is as expected.
			rules, _, err := consul.ACL().BindingRuleList(authMethodName, namespaceQuery)
			require.NoError(err)
			require.Len(rules, 1)
			actRule, _, err := consul.ACL().BindingRuleRead(rules[0].ID, namespaceQuery)
			require.NoError(err)
			require.NotNil(actRule)
			require.Equal("Kubernetes binding rule", actRule.Description)
			require.Equal(api.BindingRuleBindTypeService, actRule.BindType)
			require.Equal("${serviceaccount.name}", actRule.BindName)
			require.Equal("serviceaccount.name!=default", actRule.Selector)

```